### PR TITLE
Add conversation Web endpoint

### PR DESCRIPTION
**Pull request by jakule**
This PR adds initial implementation of conversation endpoint that allows to store and retrieve messages related to assistant conversation.

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -3329,3 +3329,22 @@ func (c *Client) GetHeadlessAuthentication(ctx context.Context, id string) (*typ
 	}
 	return headlessAuthn, nil
 }
+
+// GetAssistantMessages retrieves a headless authentication by id.
```
Comment (zmb3) [inline api/client/client.go:3333]: headless authentication?

Comment (jakule) [reply api/client/client.go:3333]: copy -> paste -> update. The last failed. Fixing. Thanks

File: lib/auth/auth.go [hunk 2]
```diff
@@ -4887,6 +4887,16 @@ func (a *Server) GetHeadlessAuthentication(ctx context.Context, name string) (*t
 	return headlessAuthn, trace.Wrap(err)
 }
 
+// GetAssistantMessages returns all messages with given conversation ID.
+func (a *Server) GetAssistantMessages(ctx context.Context, id string) (*proto.GetAssistantMessagesResponse, error) {
+	return a.Services.GetAssistantMessages(ctx, id)
```
Comment (xacrimon) [inline lib/auth/auth.go:4892]: should this be `trace.Wrap`ed?

File: lib/auth/auth_with_roles.go [hunk 3]
```diff
@@ -5812,6 +5813,16 @@ func (a *ServerWithRoles) UpdateHeadlessAuthenticationState(ctx context.Context,
 	return trace.Wrap(err)
 }
 
+func (a *ServerWithRoles) GetAssistantMessages(ctx context.Context, id string) (*proto.GetAssistantMessagesResponse, error) {
+	// TODO(jakule): Check if user has access to given conversation ID
```
Comment (xacrimon) [inline lib/auth/auth_with_roles.go:5817]: good call, we can impl RBAC checks in a future PR

File: lib/services/identity.go [hunk 4]
```diff
@@ -267,6 +267,10 @@ type Identity interface {
 	// DeleteHeadlessAuthentication deletes a headless authentication from the backend by name.
 	DeleteHeadlessAuthentication(ctx context.Context, name string) error
 
+	GetAssistantMessages(ctx context.Context, id string) (*proto.GetAssistantMessagesResponse, error)
+
+	CreateAssistantMessage(ctx context.Context, msg *proto.AssistantMessage) error
+
```
Comment (xacrimon) [inline lib/services/identity.go:273]: Can we move the comments on the other signature declarations here as well? Applies here as well as all other declarations that are missing godocs.

**Review by xacrimon — APPROVED** [review]
lgtm. ill post my changes to the API handler when this gets merged.

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 5]
```diff
@@ -2245,8 +2245,34 @@ message UpdateHeadlessAuthenticationStateRequest {
   MFAAuthenticateResponse mfa_response = 3;
 }
 
+message AssistantRequest {
+  // conversation_id used to tie all messages in a one conversation.
+  string conversation_id = 1;
+}
+
+message AssistantMessage {
```
[the remark this answers is not in the document]
Comment (jakule) [reply api/proto/teleport/legacy/client/proto/authservice.proto:2253]: Added `created_time`.

**Pull request merged**